File: MPP/MPP_Iteratia3_Java/Networking/src/main/java/AbstractServer.java
import java.io.IOException;
import java.net.ServerSocket;
import java.net.Socket;

public abstract class AbstractServer {
    private int port;
    private ServerSocket server = null;
    public AbstractServer(int port){
        this.port = port;
    }
    public void start(){
        try{
            server = new ServerSocket(port);
            while(true){
                Socket client = server.accept();
                System.out.println("Client connected...");
                processRequest(client);
            }
        } catch (IOException e) {
            e.printStackTrace();
        } finally{
            stop();
        }
    }

    public void stop(){
        try{
            server.close();
        }catch (Exception e){
            e.printStackTrace();
        }
    }

    public abstract void processRequest(Socket client);
}

File: MPP/MPP_Iteratia3_C#/AbstractServer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Sockets;
using System.Text;
using System.Threading.Tasks;

namespace Server
{
    public abstract class AbstractServer
    {
        private TcpListener server;
        private string host;
        private int port;

        public AbstractServer(string host, int port)
        {
            this.host = host;
            this.port = port;
        }

        public void Start()
        {
            IPAddress adr = IPAddress.Parse(host);
            IPEndPoint ep = new IPEndPoint(adr, port);
            server = new TcpListener(ep);
            server.Start();
            while (true)
            {
                TcpClient client = server.AcceptTcpClient();
                processRequest(client);
            }
        }

        public abstract void processRequest(TcpClient client);
    }
}
